pkg/apis/duck/v1/delivery_types.go

```diff
@@ -37,6 +38,15 @@ type DeliverySpec struct {
 	// +optional
 	Retry *int32 `json:"retry,omitempty"`
 
+	// Timeout is the timeout of each single request.
+	// More information on Duration format:
+	//  - https://www.iso.org/iso-8601-date-and-time-format.html
+	//  - https://en.wikipedia.org/wiki/ISO_8601
+	//
+	// Note: This API is EXPERIMENTAL and might break anytime. For more details: https://github.com/knative/eventing/issues/5148
+	// +optional
+	Timeout *string `json:"timeout,omitempty"`
```

Comment:
Should we clarify what 0 means and what's the default value for this setting?

Reply:
> Should we clarify what 0

Mhh I didn't thought to 0 TBH, I guess that means no timeout? But I think it's not very safe to don't have a timeout, so I prefer to disallow it.

> what's the default value

About the default value, I think we have no specification today about it (aka it's implementation dependant), so i prefer to leave it unspecified.

Reply:
Check the last commit